Pass one NDM to backend init

lucylq opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #10529
* __->__ #10528

Take external NDM if it exists, otherwise internal.

Note: by default, xnnpack uses the named_data_map. Constants are not stored with the delegated blob anymore.

Differential Revision: [D73679710](https://our.internmc.facebook.com/intern/diff/D73679710/)

lucylq committed:
Pass one NDM to backend init

Take external NDM if it exists, otherwise internal.

Note: by default, xnnpack uses the named_data_map. Constants are not stored with the delegated blob anymore.

Differential Revision: [D73679710](https://our.internmc.facebook.com/intern/diff/D73679710/)

[ghstack-poisoned]

JacobSzwejbka commented on runtime/executor/method.cpp:
```diff
@@ -805,6 +805,21 @@ Error Method::init(
+    ET_CHECK_OR_RETURN_ERROR(
+        (pte_map_items == 0 || named_data_map_items == 0),
+        NotImplemented,
+        "NamedDataMap merge not supported. Both pte_data_map and named_data_map are non-empty, don't know what to do.");
```
nit: remove the dont know what to do part haha. Direct people seeing this crash to open an issue

JacobSzwejbka commented on runtime/executor/method.cpp:
```diff
@@ -805,6 +805,21 @@ Error Method::init(
       pte_data_map = pte_data_map_res.get();
     }
 
+    int named_data_map_items =
```
we shouldnt use raw int anywhere. size_t or a fixed size prim like int32

lucylq committed:
Update on "Pass one NDM to backend init"


Take external NDM if it exists, otherwise internal.

Note: by default, xnnpack uses the named_data_map. Constants are not stored with the delegated blob anymore.

Differential Revision: [D73679710](https://our.internmc.facebook.com/intern/diff/D73679710/)

[ghstack-poisoned]

lucylq committed:
Update on "Pass one NDM to backend init"


Take external NDM if it exists, otherwise internal.

Note: by default, xnnpack uses the named_data_map. Constants are not stored with the delegated blob anymore.

Differential Revision: [D73679710](https://our.internmc.facebook.com/intern/diff/D73679710/)

[ghstack-poisoned]

lucylq committed:
Update on "Pass one NDM to backend init"


Take external NDM if it exists, otherwise internal.

Note: by default, xnnpack uses the named_data_map. Constants are not stored with the delegated blob anymore.

Differential Revision: [D73679710](https://our.internmc.facebook.com/intern/diff/D73679710/)

[ghstack-poisoned]

lucylq committed:
Update on "Pass one NDM to backend init"


Take external NDM if it exists, otherwise internal.

Note: by default, xnnpack uses the named_data_map. Constants are not stored with the delegated blob anymore.

Differential Revision: [D73679710](https://our.internmc.facebook.com/intern/diff/D73679710/)

[ghstack-poisoned]

lucylq committed:
Update on "Pass one NDM to backend init"


Take external NDM if it exists, otherwise internal.

Note: by default, xnnpack uses the named_data_map. Constants are not stored with the delegated blob anymore.

Differential Revision: [D73679710](https://our.internmc.facebook.com/intern/diff/D73679710/)

[ghstack-poisoned]

lucylq committed:
Update on "Pass one NDM to backend init"


Take external NDM if it exists, otherwise internal.

Note: by default, xnnpack uses the named_data_map. Constants are not stored with the delegated blob anymore.

Differential Revision: [D73679710](https://our.internmc.facebook.com/intern/diff/D73679710/)

[ghstack-poisoned]

lucylq committed:
Update on "Pass one NDM to backend init"


Take external NDM if it exists, otherwise internal.

Note: by default, xnnpack uses the named_data_map. Constants are not stored with the delegated blob anymore.

Differential Revision: [D73679710](https://our.internmc.facebook.com/intern/diff/D73679710/)

[ghstack-poisoned]
